src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java

```diff
@@ -271,562 +268,566 @@ private static String getCustomURI(String storageAccountName, String type,
+                            }
+                        }
+                        fileName = fileName.substring(0, embVPSepIndex);
+                    }
+                }
+
+                archiveIncludes += "," + fileName;
+
+                // List all the paths without the zip archives.
+                FilePath[] paths = workspacePath.list(fileName, excludesWithoutZip);
+                filesUploaded += paths.length;
+
+                URI workspaceURI = workspacePath.toURI();
+
+                if (paths.length != 0) {
+                    if (uploadType != UploadType.ZIP) {
+                        for (FilePath src : paths) {
+                            // Remove the workspace bit of this path
+                            URI srcURI = workspaceURI.relativize(src.toURI());
+                            CloudBlockBlob blob = null;
+                            String srcPrefix = srcURI.getPath();
+                            if (Utils.isNullOrEmpty(expVP)
+                                    && Utils.isNullOrEmpty(embeddedVP)) {
+                                blob = container.getBlockBlobReference(srcPrefix);
+                            } else {
+                                String prefix = expVP;
+
+                                if (!Utils.isNullOrEmpty(embeddedVP)) {
+                                    if (Utils.isNullOrEmpty(expVP)) {
+                                        prefix = embeddedVP;
+                                    } else {
+                                        prefix = expVP + embeddedVP;
+                                    }
+                                }
+                                blob = container.getBlockBlobReference(prefix + srcPrefix);
+                            }
+
+                            upload(listener, blob, src);
+
+                            individualBlobs.add(new AzureBlob(blob.getName(), blob.getUri().toString().replace("http://", "https://")));
+                        }
+                    }
+                }
+            }
+
+            if (filesUploaded != 0 && (uploadType != UploadType.INDIVIDUAL)) {
+                // Create a temp dir for the upload
+                FilePath tempPath = workspacePath.createTempDir(zipFolderName, null);
+
+                Glob globScanner = new Glob(archiveIncludes, excludesWithoutZip);
+
+                FilePath zipPath = tempPath.child(zipName);
+                workspacePath.zip(zipPath.write(), globScanner);
+
+                // When uploading the zip, do not add in the tempDir to the block
+                // blob reference.
+                String blobURI = zipPath.getName();
+
+                if (!Utils.isNullOrEmpty(expVP)) {
+                    blobURI = expVP + blobURI;
+                }
+
+                CloudBlockBlob blob = container.getBlockBlobReference(blobURI);
+
+                telemetry.createEvent("AzureJenkinsStorageUploadEvent", filesUploaded + " files uploaded, upload type " + uploadType.toString());
```

Comment:
right now there is no implementation in the code  that keep tracks of blob size, we can implement it. will wait for Arun ...